src/editor.ts

```diff
@@ -335,4 +359,20 @@ export class Editor {
 		vscode.commands.executeCommand("emacs.cursorHome");
 		vscode.commands.executeCommand("emacs.cursorDown");
 	}
+
+	enterMarkMode() :void {
+		if (Editor.inMarkMode && !Editor.markHasMoved) {
+			this.exitMarkMode();
+		} else {
+			this.positions.push(vscode.window.activeTextEditor.selection.active)
+			this.exitMarkMode();
+			Editor.inMarkMode = true;
+			Editor.markHasMoved = false;
+		}
+	}
+
+	exitMarkMode() : void {
+		vscode.commands.executeCommand("cancelSelection");
+		Editor.inMarkMode = false;
+	}
 }
```

Comment:
It makes more sense and allows interesting things to be coded more easily to have these functions here instead of in extensions.ts file